linalg/hypre.cpp

```diff
@@ -1266,8 +1266,20 @@ HypreParMatrix::HypreParMatrix(MPI_Comm comm, int nrows,
    }
 
    // Copy in the row and column partitionings
+   const bool rows_eq_cols = [rows,cols,part_size]()
```

Comment:
It is best to `MPI_Allreduce` this value with logical AND, i.e. this Boolean means that the partitionings are globally equal -- this way all ranks will make the same decisions in the following code.

Reply:
OK, done in 8be0dee00.